csharp/src/Drivers/Apache/Hive2/HiveServer2Reader.cs

```diff
@@ -78,6 +105,55 @@ static IArrowArray GetArray(TColumn column)
                 (IArrowArray?)column.StringVal?.Values ??
                 (IArrowArray?)column.BinaryVal?.Values ??
                 throw new InvalidOperationException("unsupported data type");
+            if (expectedArrowType != null && arrowArray is StringArray stringArray && s_arrowStringConverters.ContainsKey(expectedArrowType.TypeId))
+            {
+                // Perform a conversion from string to native/scalar type.
+                Func<StringArray, IArrowType, IArrowArray> converter = s_arrowStringConverters[expectedArrowType.TypeId];
+                return converter(stringArray, expectedArrowType);
+            }
+            return arrowArray;
+        }
+
+        private static Date32Array ConvertToDate32(StringArray array, IArrowType _)
+        {
+            var resultArray = new Date32Array.Builder();
+            foreach (string item in (IReadOnlyCollection<string>)array)
+            {
+                resultArray.Append(DateTime.Parse(item));
```

Comment:
Does the date come back in a predictable format which could be parsed more efficiently than by DateTime.Parse? (e.g. if we know it always comes back as yyyy-mm-dd?). DateTime.Parse handles lots of cases that we probably don't need to support.

Otherwise, this should probably have CultureInfo.InvariantCulture passed to it.

Reply:
Now checking for a known format.